src/stylesheet/index.tsx

```diff
@@ -0,0 +1,20 @@
+import { ExtendedTheme } from "@constants/themes";
+import { useTheme } from "@react-navigation/native";
+import { useMemo } from "react";
+import { StyleSheet } from "react-native";
+
+type Colors = ExtendedTheme["colors"];
+
+export const useStylesWithTheme = <T extends StyleSheet.NamedStyles<T>>(
+  stylesheet: (colors: Colors) => T
```

Comment:
nit: sometimes you use `styleSheet`, sometimes `stylesheet`

Reply:
can't find it styleSheet 🤔 maybe in test pr